File: src/leetCode/Ex34.java
package leetCode;

import java.util.Arrays;

public class Ex34 {
    public int[] searchRange(int[] nums, int target) {
        int l = helper(nums, target-0.5f);
        int r = helper(nums, target+0.5f);
        if(l==r){
            return new int[]{-1,-1};
        }else{
            return new int[]{l, r-1};
        }
    }

    private int helper(int[] nums, float target){
        int l = 0;
        int r = nums.length;
        while(l<r){
            int mid = (r-l)/2+l;
            if(nums[mid]<target){
                l= mid+1;
            }else{
                r = mid;
            }
        }
        return l;
    }
    public static void main(String[] args) {
        var o = new Ex34();
        // var res = o.searchRange(new int[]{5,7,7,8,8,10}, 8);
        var res = o.searchRange(new int[]{}, 6);        
        // var res = o.searchRange(new int[]{5,7,7,8,8,10}, 6);        

        System.out.println(Arrays.toString(res));
    }
}

File: src/LeetCode/ex34.kt
package LeetCode

fun searchRange(nums: IntArray, target: Int): IntArray {
    if (nums.isEmpty()){
        return intArrayOf(-1,-1)
    }
    var left = 0
    var right = nums.size
    while (left<right){
        val mid = (left+right)/2
        if (nums[mid]<target) {
            left=mid+1
        }
        else {
            right=mid
        }
    }
    if (left>=nums.size)
        return intArrayOf(-1,-1)
    val start= if (nums[left]==target) left else return intArrayOf(-1,-1)
    right=nums.size
    while (left<right){
        val mid = (left+right)/2
        if (nums[mid]>target) {
            right=mid
        }
        else {
            left=mid+1
        }
    }
    return intArrayOf(start,left-1)
}

fun main(args: Array<String>) {
    searchRange(intArrayOf(1, 2, 3, 3, 3, 4, 5), 3).forEach{ println(it)}
}